bin/yarn

```diff
@@ -3,7 +3,7 @@ argv0=$(echo "$0" | sed -e 's,\\,/,g')
 basedir=$(dirname "$(readlink "$0" || echo "$argv0")")
 
 case "$(uname -s)" in
-  Darwin) basedir="$( cd "$( dirname "$argv0" )" && pwd )";;
+  Darwin) basedir="$(cd "$(dirname "$0")" && cd "$basedir" && pwd)";;
```

Comment:
This diff doesn't make sense to we. Why are you calling `cd` twice? It looks like it makes this line become a noop.

Reply:
This first forces the current directory to be the same as the binary (example: /usr/local/bin), which is the directory the symlink are relative to. Then, the symlink (example: ../share/yarn/bin) can be resolved and entered.

Reply:
This is all necessary because the macOS shell comes with no native "path.join()"

Reply:
Shrinking it to `cd "$(dirname "$0")$basedir"` will break cases where the symlink is to an absolute path (example: /usr/local/bin/usr/local/share/yarn/bin/)